### ISSUE TITLE
Error when running in parallel

### ISSUE THREAD

**Issue by edwjusti** [pestphp/pest#859]
We've encountered a few errors when running pest in parallel with coverage collection AND datasets. 

Sample command:
`
./vendor/bin/pest --group=unit --parallel --log-junit=coverage/report.xml  --colors=never --coverage --coverage-cobe
rtura=coverage/coverage.cobertura.xml
`

When running without --filter I get the following error:
[Error 1](https://pastebin.com/DN5SGbJr)

However when running with --filter on any file (including those with those without datasets), I get another error:
[Error 2](https://pastebin.com/CgfuwSpz)

**Comment by keyvangholami** [comment]
The error you are encountering is an ArgumentCountError and is being thrown because the function PHPUnit\Runner\Filter\GroupFilterIterator::__construct() is expecting exactly 3 arguments but only 1 is being passed.

Looking at the stack trace, this error occurs when trying to construct the GroupFilterIterator within the RecursiveFilterIterator->getChildren() method.

**Comment by devajmeireles** [comment]
Hey, @edwjusti . Thank you for your report.

Since you already got an answer above and you didn't answer anymore, I will understand that you managed to figure out the problem. Furthermore, we have not received any reports similar to this, indicating that the problem maybe is not with Pest.

Due organization, I will close this issue. Feel free to open a new issue whenever necessary. Remember to fill in as many details as possible.
